content/post/2025/02-riscv-formal-extension-for-vc-formal.md

```diff
@@ -0,0 +1,298 @@
+- **Enable Easy `.sby → .tcl` Conversion** – Support straightforward translation of `.sby` configurations into `.tcl` scripts, even outside the framework.  
+
+By achieving these goals, the project enhances the **efficiency, accessibility, and adoption** of formal verification for **RISC-V cores**, fostering wider collaboration in the verification community.  
+
+## Adapting riscv-formal for VC Formal & The Verification Process  
+
+Integrating the **riscv-formal framework** with **Synopsys VC Formal** required overcoming several technical challenges, given the differences between **open-source** and **proprietary** formal verification tools.  
+
+This section outlines the core modifications and methodology used to **bridge the gap** between these two verification environments.  
+
+
+## 1. Key Challenges in Transitioning to VC Formal  
+
+- **Conversion from SymbiYosys Flow:** The **.sby** configuration files used in riscv-formal are specific to SymbiYosys. VC Formal, on the other hand, operates using **Tcl-based** scripts, requiring a structured **translation mechanism** to adapt the verification flow.  
+- **Environment Setup & Automation:** Automating the process—so that users can easily set up **VC Formal verification** with minimal manual intervention—was a key goal in this adaptation.  
+
+## 2. Verification Process  
+
+### i. FOSS SymbiYosys Formal Verification Flow
+
+This flow demonstrates the formal verification process using the riscv-formal framework with the open-source SymbiYosys tool. It consists of components like RVFI, the wrapper, and `checks.cfg`. These components are used to generate `.sby` files, which are then processed by SymbiYosys to run verification checks and produce the results, ensuring that the RISC-V implementation adheres to formal specifications. This flow represents a streamlined, FOSS-based approach to formal verification.
+
+![FOSS SymbiYosys formal verification flow](/static/static-2025/Sby.png)
+
+### ii. Synopsys VC Formal Verification Flow
+
+This flow illustrates the modified process, where the riscv-formal framework is used with Synopsys VC Formal. The `.sby` files from the riscv-formal framework are first converted to `.tcl` scripts using a custom `sby to tcl` converter before being processed by VC Formal to generate verification results.
+
+![Synopsys VC Formal verification flow](/static/static-2025/VCF.png)
+
+The framework consists of **three key stages**, each automated for smooth verification:  
+
+1. **Pre-Processing:** Converts `.sby` files into `.tcl` scripts for **VC Formal compatibility**.  
+2. **Processing:** Runs the **formal checks**, collects logs, and categorizes **warnings & errors**.  
+3. **Post-Processing:** Summarizes verification **results**, helping users quickly identify pass/fail conditions.  
+
+To achieve this, we’ve extended the framework with four core components:  
+
+### **1️⃣ sby_to_tcl.py – Automating Test Setup (Pre-Processing)**  
```

Comment:
Does this need to be bolded?  It's already a heading.  Same for the following headings.